src/unit_tests/logcollector/test_logcollector.c

```diff
@@ -40,140 +40,194 @@ void w_initialize_file_status();
 
     expect_string(__wrap_OS_SHA1_File_Nbytes_with_fp_check, fname, lf->file);
-    expect_value(__wrap_OS_SHA1_File_Nbytes_with_fp_check, mode, mode);
+    expect_value(__wrap_OS_SHA1_File_Nbytes_with_fp_check, mode, OS_BINARY);
     expect_value(__wrap_OS_SHA1_File_Nbytes_with_fp_check, nbytes, position);
     expect_value(__wrap_OS_SHA1_File_Nbytes_with_fp_check, fd_check, 0);
     will_return(__wrap_OS_SHA1_File_Nbytes_with_fp_check, "32bb98743e298dee0a654a654765c765d765ae80");
     will_return(__wrap_OS_SHA1_File_Nbytes_with_fp_check, -1);
 
-    bool ret = w_get_hash_context (lf, context, position);
+    bool ret = w_get_hash_context (lf, NULL, position);
 
     assert_false(ret);
-
-    os_free(context);
 }
 
 void test_w_get_hash_context_done(void ** state) {
-
-    SHA_CTX * context;
-    os_calloc(1, sizeof(SHA_CTX), context);
     int64_t position = 10;
+    test_logcollector_t *test_struct = *state;
 
-    logreader *lf = NULL;
-    os_calloc(1, sizeof(logreader), lf);
-    lf->fp = (FILE*)1;
-    os_strdup("/test_path", lf->file);
+    logreader *lf = test_struct->log_reader;
+    SHA_CTX *context = test_struct->context;
+    os_file_status_t *data = test_struct->status;
 
-    os_file_status_t data = {0};
-    data.context.num = 123;
+    lf->file = strdup("/test_path");
+    data->context.num = 123;
 
-    expect_any(__wrap_OSHash_Get_ex, self);
-    expect_string(__wrap_OSHash_Get_ex, key, lf->file);
-    will_return(__wrap_OSHash_Get_ex, &data);
+    expect_function_call_any(__wrap_pthread_rwlock_rdlock);
+    expect_function_call_any(__wrap_pthread_rwlock_unlock);
+    OSHash_Add(files_status, lf->file, data);
 
-    bool ret = w_get_hash_context (lf, context, position);
+    bool ret = w_get_hash_context(lf, context, position);
 
-    assert_memory_equal(&(data.context), context, sizeof(SHA_CTX));
+    assert_memory_equal(&(data->context), context, sizeof(SHA_CTX));
     assert_true(ret);
-
-    os_free(context);
+    // The status should be freed.
+    test_struct->status = NULL;
 }
 
-/* w_update_file_status */
-
+/* w_update_file_status
+Can't test if OSHashAddEx isn't wrapped, but wrapping it will leak memory ;(
 void test_w_update_file_status_fail_update_add_table_hash(void ** state) {
     test_mode = 1;
 
     char * path = "test/test.log";
     long pos = 0;
     SHA_CTX context = {0};
 
+    expect_function_call_any(__wrap_pthread_rwlock_wrlock);
+    expect_function_call_any(__wrap_pthread_rwlock_unlock);
     expect_value(__wrap_OS_SHA1_Stream, buf, NULL);
     will_return(__wrap_OS_SHA1_Stream, "a7a899f25aeda32989d1029839ef2e594835c211");
 
-    will_return(__wrap_OSHash_Update_ex, 0);
-
-    OSHash_Add_ex_check_data = 0;
-    expect_value(__wrap_OSHash_Add_ex, self, files_status);
-    expect_string(__wrap_OSHash_Add_ex, key, path);
-    will_return(__wrap_OSHash_Add_ex, 0);
-
     int retval = w_update_file_status(path, pos, &context);
-
-    assert_int_equal(retval,-1);
-
+    assert_int_equal(retval, -1);
 }
```

Comment:
We should not do any extra changes to tests that need to revisited other than comment them out, please leave the function as it was and comment it out.